Server/JokesOnYou.Web.Api/Services/JokesService.cs

```diff
@@ -26,5 +26,29 @@ public async Task<IEnumerable<JokeReplyDto>> GetAllJokeDtosAsync()
 
             return jokeDtos;
         }
+
+        public async Task UpdateJoke(JokeUpdateDTO jokeUpdateDTO)
+        {
+            // get the joke we need to update
+            // set fields with jokeUpdateDTO
+            var jokeToUpdate = await _jokesRepo.GetJokeToUpdate(jokeUpdateDTO);
+
+            if (jokeToUpdate == null)
+            {
+                throw new NotImplementedException("can't find joke to update");
+            }
+
+            jokeToUpdate.Premise = jokeUpdateDTO.Premise;
```

Comment:
We can use automapper here. So instead of mapping to a new item we can map to an existing

it would look like this map(From, To)